smctl/main.go

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright 2018 The Service Manager Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+package main
+
+import (
+	"github.com/Peripli/service-manager-cli/internal/cmd"
+	"github.com/Peripli/service-manager-cli/internal/cmd/broker"
+	"github.com/Peripli/service-manager-cli/internal/cmd/info"
+	"github.com/Peripli/service-manager-cli/internal/cmd/login"
+	"github.com/Peripli/service-manager-cli/internal/cmd/platform"
+	"github.com/Peripli/service-manager-cli/internal/cmd/version"
+
+	"os"
+)
+
+func main() {
+	clientVersion := "0.0.1"
+
+	context := &cmd.Context{}
+	rootCmd := cmd.BuildRootCommand(context)
+
+	commands := []cmd.CommandWrapper{
```

Comment:
I was thinking if we can simplify the _main.go_ file by extracting this bootstrap logic into another place, like for example in the **cmd** package into _smctl.go_ (in the same way we that we've externalized the bootstrapping of the controllers in the Service Manager).

Then the _main.go_ file would simply look like this:

    clientVersion := "0.0.1"
	context := &cmd.Context{}
	rootCmd := cmd.BuildRootCommand(context)
	cmd.BootstrapRootCommand(rootCmd)
	cmd.Execute(rootCmd)

The problem I see with this suggestion however is that a cyclic dependency would emerge this way...

Reply:
we can move the bootstrap logic to a bootstrap package.

Reply:
It's an option. But if there's nothing else in this new package other than the mentioned bootstrap function, it may not be reasonable to make a whole new package for it and we'd be better of having everything in _main.go_ like it is right now :)